execution_engine2.spec

```diff
@@ -452,18 +453,35 @@
         errormsg - string - message (e.g. stacktrace) accompanying an errored job
         error - object - the JSON-RPC error package that accompanies the error code and message
 
-            terminated_code - int - internal reason why a job was terminated, one of:
-                0 - user cancellation
-                1 - admin cancellation
-                2 - terminated by some automatic process
+        #TODO, add these to the structure?
+        condor_job_ads - dict - condor related job information
 
-            @optional error
-            @optional error_code
-            @optional errormsg
-            @optional terminated_code
-            @optional estimating
-            @optional running
-            @optional finished
+        retry_count - int - generated field based on length of retry_ids
+        retry_ids - list - list of jobs that are retried based off of this job
+        retry_parent - str - job_id of the parent this retry is based off of. Not available on a retry_parent itself
+        retry_saved_toggle - bool - Marked true when all retry steps/txns have completed
```

Comment:
Hmm, I was thinking `retry_saved_toggle` would not be exposed in the API - I'm still not sure why you'd want to do that.

Reply:
Well, if we wanted to run a script that reaped via the API, or wanted to see all jobs that had failed during a recent outage by looking for that toggle, that would be useful. Otherwise we would have to specifically hide it from the check_job call

Reply:
Or the frontend can display an error if it sees that field

Reply:
My thinking was that if a job was fetched from the DB where that toggle was `False` the server would just run a fix up routine to finish the update (at some point, not in this PR). If that's the case the toggle should never show any state other than `True` so there's no point in exposing it.

Reply:
So if I ran a check_jobs_date_range and it returned 10000 jobs, and 500 of them had failed retries, would it wait for 500 of them to run the routine? Or would it just assume they are true and not wait for the routine to run? What if the routine failed for some of the jobs, would the whole api call fail?

Reply:
> and 500 of them had failed retries, would it wait for 500 of them to run the routine?
That's what I would do, because this failure mode is likely going to be very rare and jobs with the toggle left as `False` are going to be infrequent.

Reply:
this is done

Reply:
:+1: